core/src/test/java/org/helsinki/paytrail/request/refunds/PaytrailRefundCreateTest.java

```diff
@@ -0,0 +1,199 @@
+package org.helsinki.paytrail.request.refunds;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.gson.Gson;
+import lombok.extern.slf4j.Slf4j;
+import org.helsinki.paytrail.PaytrailClient;
+import org.helsinki.paytrail.PaytrailCommonTest;
+import org.helsinki.paytrail.constants.PaymentStatus;
+import org.helsinki.paytrail.mapper.PaytrailPaymentCreateResponseMapper;
+import org.helsinki.paytrail.mapper.PaytrailRefundCreateResponseMapper;
+import org.helsinki.paytrail.model.payments.PaymentCallbackUrls;
+import org.helsinki.paytrail.model.payments.PaymentCustomer;
+import org.helsinki.paytrail.model.payments.PaymentItem;
+import org.helsinki.paytrail.model.payments.PaytrailPaymentResponse;
+import org.helsinki.paytrail.model.refunds.PaytrailRefundResponse;
+import org.helsinki.paytrail.model.refunds.RefundItem;
+import org.helsinki.paytrail.request.payments.PaytrailPaymentCreateRequest;
+import org.helsinki.paytrail.response.payments.PaytrailPaymentCreateResponse;
+import org.helsinki.paytrail.response.refunds.PaytrailRefundCreateResponse;
+import org.helsinki.paytrail.service.PaytrailSignatureService;
+import org.junit.jupiter.api.Test;
+
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.TreeMap;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@Slf4j
+class PaytrailRefundCreateTest extends PaytrailCommonTest {
+
+    private PaytrailRefundCreateResponseMapper refundMapper;
+    private PaytrailPaymentCreateResponseMapper paymentMapper;
+
+    public PaytrailRefundCreateTest() {
+        this.refundMapper = new PaytrailRefundCreateResponseMapper(new ObjectMapper());
+        this.paymentMapper = new PaytrailPaymentCreateResponseMapper(new ObjectMapper());
+    }
+
+    @Test
+    public void createTestRefundNormalMerchant() throws ExecutionException, InterruptedException {
+        PaytrailClient client = new PaytrailClient(merchantId, secretKey);
+
+        /* Creating a test payment */
+        PaytrailPaymentResponse paymentResponse = createTestNormalMerchantPayment(client);
+
+        PaytrailRefundCreateRequest.CreateRefundPayload payload = new PaytrailRefundCreateRequest.CreateRefundPayload();
+
+        String refundStamp = UUID.randomUUID().toString();
+        String refundReference = "3759170";
+
+        payload.setRefundStamp(refundStamp);
+        payload.setRefundReference(refundReference);
+        payload.setAmount(10);
+        payload.setEmail("martin.lehtomaa@ambientia.fi");
```

Comment:
I think you should change this :D

Reply:
Hahaha true :D I think all the tests have these left, I will remove it from all of them 👍